src/github_runner_manager/openstack_cloud/openstack_runner_manager.py

```diff
@@ -636,7 +639,15 @@ def _wait_runner_startup(self, instance: OpenstackInstance) -> None:
                 "not completed"
             ) from err
 
-        result: invoke.runners.Result = ssh_conn.run("ps aux", warn=True)
+        result: invoke.runners.Result = ssh_conn.run("cloud-init status", warn=True)
+        if not result.ok:
+            logger.warning(
+                "cloud-init status command failed on %s: %s.", instance.server_name, result.stderr
+            )
+            raise RunnerStartError(f"Runner startup process not found on {instance.server_name}")
+        if CloudInitStatus.DONE in result.stdout:
+            return
```

Comment:
Maybe add a little comment for other devs to provide reasoning on why we accept it

Reply:
Added comment, thank you!